engines/teenagent/teenagent.cpp

```diff
@@ -1141,4 +1146,55 @@ void TeenAgentEngine::setTTSVoice(CharacterID characterID) const {
 	}
 }
 
+Common::U32String TeenAgentEngine::convertCyrillic(const Common::String &text) const {
+	const byte *bytes = (const byte *)text.c_str();
+	static byte result[10240];
```

Comment:
I am not fond of having a 10k static variable, which means adding 10k to the executable data segment. This could be an issue for memory restricted platform. Those are unlikely to have TTS, but the issue is that your function is currently always present.